src/Workleap.Extensions.Http.Authentication.ClientCredentialsGrant/ClientCredentialsTokenManagementService.cs

```diff
@@ -59,4 +65,38 @@ private async Task<ClientCredentialsToken> SynchronizeAsync(string clientName, F
             this._lazyGetNewTokenTasks.TryRemove(clientName, out _);
         }
     }
+
+    private Func<Task<ClientCredentialsToken>> GetNewTokenTaskFactory(string clientName, CancellationToken cancellationToken) => async () =>
+    {
+        var newToken = await this._tokenEndpointService.RequestTokenAsync(clientName, cancellationToken).ConfigureAwait(false);
+
+        var cacheEvictionTime = await this._tokenCache.SetAsync(clientName, newToken, cancellationToken).ConfigureAwait(false);
+        var cacheDuration = cacheEvictionTime - DateTimeOffset.UtcNow;
+
+        if (cacheDuration > TimeSpan.Zero)
+        {
+            const double backgroundRefreshDelayFactor = 0.8;
+            var delayBeforeNextBackgroundRefresh = TimeSpan.FromTicks((long)Math.Round(cacheDuration.Ticks * backgroundRefreshDelayFactor));
+
+            _ = this.ScheduleTokenBackgroundRefreshAsync(clientName, delayBeforeNextBackgroundRefresh);
```

Comment:
Same comment here about Fire and Forget. If `_backgroundRefreshCancellationToken` is cancelled while `Task.Delay`, you get an exception, so an TaskUnhandledException somewhere.

Reply:
Done